server/etcdserver/v3_server.go

```diff
@@ -353,8 +353,22 @@ func (s *EtcdServer) LeaseRenew(ctx context.Context, id lease.LeaseID) (int64, e
 		if !s.ensureLeadership() {
 			return -1, lease.ErrNotPrimary
 		}
-		if err := s.waitAppliedIndex(); err != nil {
-			return 0, err
+
+		// This change aims to make lease renewal faster under high server load
+		// while preserving correctness. If a lease is not found, it might still be in
+		// the process of being created. We must wait for the applied index to advance
+		// to verify whether the lease truly does not exist.
+		if s.FeatureEnabled(features.FastLeaseKeepAlive) {
+			le := s.lessor.Lookup(id)
+			if le == nil {
+				if err := s.waitAppliedIndex(); err != nil {
```

Comment:
I tried to reproduce that issue by defragging 1GiB database.
It's kind of stop-the-world. In this case, it's still unable to refresh TTL.

Reply:
Because defragment blocks applying. This may cause `s.waitAppliedIndex()` block until defragment finishes.